plugins/parquet/parquetembed.cpp

```diff
@@ -838,6 +838,125 @@ int64_t ParquetRowBuilder::currArrayIndex()
+            return serialized.str();
+        case BinaryType:
+            return (*array_visitor)->bin_arr->GetView(currArrayIndex());
+        case LargeBinaryType:
+            return (*array_visitor)->large_bin_arr->GetView(currArrayIndex());
+        case DoubleType:
+            tokenSerializer.serialize((*array_visitor)->double_arr->Value(currArrayIndex()), serialized);
+            return serialized.str();
+        case HalfFloatType:
+            tokenSerializer.serialize((*array_visitor)->half_float_arr->Value(currArrayIndex()), serialized);
+            return serialized.str();
+        case FloatType:
+            tokenSerializer.serialize((*array_visitor)->float_arr->Value(currArrayIndex()), serialized);
+            return serialized.str();
+        case IntType:
+            tokenSerializer.serialize(getSigned(array_visitor, currArrayIndex()), serialized);
+            return serialized.str();
+        case UIntType:
+            tokenSerializer.serialize(getUnsigned(array_visitor, currArrayIndex()), serialized);
+            return serialized.str();
+        case DateType:
+            tokenSerializer.serialize((*array_visitor)->size == 32 ? (__int32) (*array_visitor)->date32_arr->Value(currArrayIndex()) : (__int64) (*array_visitor)->date64_arr->Value(currArrayIndex()), serialized);
+            return serialized.str();
+        case TimestampType:
+            tokenSerializer.serialize((__int64) (*array_visitor)->timestamp_arr->Value(currArrayIndex()), serialized);
+            return serialized.str();
+        case TimeType:
+            tokenSerializer.serialize((*array_visitor)->size == 32 ? (__int32) (*array_visitor)->time32_arr->Value(currArrayIndex()) : (__int64) (*array_visitor)->time64_arr->Value(currArrayIndex()), serialized);
+            return serialized.str();
+        case DurationType:
+            tokenSerializer.serialize((__int64) (*array_visitor)->duration_arr->Value(currArrayIndex()), serialized);
+            return serialized.str();
+        case StringType:
+            return (*array_visitor)->string_arr->GetView(currArrayIndex());
+        case LargeStringType:
+            return (*array_visitor)->large_string_arr->GetView(currArrayIndex());
+        case DecimalType:
+            return (*array_visitor)->size == 128 ? (*array_visitor)->dec_arr->GetView(currArrayIndex()) : (*array_visitor)->large_dec_arr->GetView(currArrayIndex());
+        default:
+            failx("Unimplemented Parquet type for field with name %s.", field->name);
+    }
+}
+
+__int64 ParquetRowBuilder::getCurrValue(const RtlFieldInfo *field)
+{
+    switch ((*array_visitor)->type)
+    {
+        case BoolType:
+            return (*array_visitor)->bool_arr->Value(currArrayIndex());
+        case IntType:
+            return getSigned(array_visitor, currArrayIndex());
+        case UIntType:
+            return getUnsigned(array_visitor, currArrayIndex());
+        case DoubleType:
```

Comment:
Potential for a getReal() function similar to getSigned/unsigned which returned values for Double/HalfFloat/Float which is also called from getRealResult()

Reply:
Added a getReal() function and removed extra types. Added getReal to getCurrIntValue() and getRealResult().